src/Compilers/CSharp/Portable/Binder/BindingDiagnosticBag.cs

```diff
@@ -203,5 +148,34 @@ protected override bool ReportUseSiteDiagnostic(DiagnosticInfo diagnosticInfo, D
         {
             return Symbol.ReportUseSiteDiagnostic(diagnosticInfo, diagnosticBag, location);
         }
+
+        internal CSDiagnosticInfo Add(ErrorCode code, Location location)
+        {
+            var info = new CSDiagnosticInfo(code);
+            DiagnosticBag?.Add(new CSDiagnostic(info, location));
```

Comment:
>DiagnosticBag?.Add(new CSDiagnostic(info, location)) [](start = 12, length = 52)

Consider replacing with `Add(info, location)` here and in the two methods below.